File: app-hungry-hungry-hip-ai/src/app/modules/core/domain/restaurant.ts
export interface Restaurant {
  id: number;
  address: string;
  name: string;
  menuCategories: MenuCategory[];
}

export interface MenuCategory {
  id: number;
  name: string;
  menuItems: MenuItem[];
}

export interface MenuItem {
  id: number;
  name: string;
  description: string;
  price: number;
}

File: service-hungry-hungry-hip-ai/Restaurants.API/Domain/Restaurant.cs
namespace Restaurants.API.Domain
{
	public class Restaurant
	{
		public int Id { get; set; }
		public string Name { get; set; } = string.Empty;
		public string Address { get; set; } = string.Empty;
		public List<MenuCategory> MenuCategories { get; set; } = new List<MenuCategory>();
	}

	public class MenuCategory
	{
		public int Id { get; set; }
		public string Name { get; set; } = string.Empty;
		public List<MenuItem> MenuItems { get; set; } = new List<MenuItem>();
	}

	public class MenuItem
	{
		public int Id { get; set; }
		public string Name { get; set; } = string.Empty;
		public string Description { get; set; } = string.Empty;
		public decimal Price { get; set; }
	}
}
